runner/utils/driver/istio/istio.go

```diff
@@ -219,6 +220,146 @@ func (i *Istio) ApplyService(service *model.Service) error {
 	return nil
 }
 
+func (i *Istio) DeleteService(serviceId, projectId, version string) error {
+	service := &model.Service{ID: serviceId, ProjectID: projectId, Version: version}
+	err := i.kube.CoreV1().ServiceAccounts(projectId).Delete(getServiceAccountName(service), &metav1.DeleteOptions{})
+	if kubeErrors.IsNotFound(err) {
+		// service account not found meaning no service present
+		logrus.Errorf("error deleting service in istio service account not found got error message - %v", err)
+		return err
+	} else if err == nil {
+		if err = i.kube.AppsV1().Deployments(projectId).Delete(getDeploymentName(service), &metav1.DeleteOptions{}); err != nil {
+			logrus.Errorf("error deleting service in istio unable to find deployment got error message - %v", err)
+			return err
+		}
+		if err = i.kube.CoreV1().Services(projectId).Delete(serviceId, &metav1.DeleteOptions{}); err != nil {
+			logrus.Errorf("error deleting service in istio unable to find services got error message - %v", err)
+			return err
+		}
+		if err = i.istio.NetworkingV1alpha3().VirtualServices(projectId).Delete(serviceId, &metav1.DeleteOptions{}); err != nil {
+			logrus.Errorf("error deleting service in istio unable to find virtual services got error message - %v", err)
+			return err
+		}
+		if err = i.istio.NetworkingV1alpha3().DestinationRules(projectId).Delete(serviceId, &metav1.DeleteOptions{}); err != nil {
+			logrus.Errorf("error deleting service in istio unable to find destination rule got error message - %v", err)
+			return err
+		}
```

Comment:
Add a comment here that when we add versioning support, the destination rules and virtual services can only be removed when all service versions are removed